lib/src/hex_util.rs

```diff
@@ -14,7 +14,17 @@
 
 #![allow(missing_docs)]
 
-fn to_reverse_hex_digit(b: u8) -> Option<u8> {
+pub fn to_reverse_hex_digit(b: u8) -> u8 {
```

Comment:
nit: might be simpler to do hex encoding by ourselves. It won't gain SIMD optimization, but the performance wouldn't matter.

Reply:
As in roll all the hex stuff ourselves and put it into this module? Fine by me

Reply:
Oh, I just meant `to_reverse_hex(&str) -> Option<String>` could be reimplemented as `encode_reverse_hex(&[u8]) -> String`.

It might be okay to self-host all hex encoding functions if that helps reduce dependency, though.